### PR TITLE
test(kubeclient): add IsReady edge-case coverage and missing-NodeReady behavior fix

### PR DESCRIPTION
**Pull request by r0hansaxena**
### Ⅰ. Describe what this PR does

`test(pkg/utils/kubeclient)` : add IsReady edge-case tests and update IsReady to return false when NodeReady is missing.

### Ⅱ. Does this pull request fix one issue?

part of #5407

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.

- `IsReady`: cover no `NodeReady` condition, `NodeReady=Unknown`, and mixed conditions where `NodeReady=False`

### Ⅳ. Describe how to verify it

go test ./pkg/utils/kubeclient -v --ginkgo.focus="IsReady"

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/utils/kubeclient/node_test.go [hunk 1]
```diff
@@ -124,4 +124,66 @@ var _ = Describe("IsReady", func() {
...
+					Conditions: []corev1.NodeCondition{
+						{
+							Type:   corev1.NodeMemoryPressure,
+							Status: corev1.ConditionFalse,
+						},
+					},
+				},
+			}
+			result := IsReady(node)
+			Expect(result).To(BeTrue())
```
[the remark this answers is not in the document]
Comment (r0hansaxena) [reply pkg/utils/kubeclient/node_test.go]: I’ve updated this by aligning `IsReady` to return `false` when `NodeReady` is missing, and adjusted the test accordingly.

File: pkg/utils/kubeclient/node.go [hunk 2]
```diff
@@ -42,11 +42,10 @@ func GetNode(client client.Reader, name string) (node *corev1.Node, err error) {
 // IsReady checks if the node is ready
 // If the node is ready,it returns True.Otherwise,it returns False.
 func IsReady(node corev1.Node) (ready bool) {
-	ready = true
+	ready = false
```
[the remark this answers is not in the document]
Comment (r0hansaxena) [reply pkg/utils/kubeclient/node.go]: I've removed the redundant named-return initialization in IsReady.

File: pkg/utils/kubeclient/node.go [hunk 3]
```diff
@@ -42,11 +42,9 @@ func GetNode(client client.Reader, name string) (node *corev1.Node, err error) {
...
-	ready = true
 	for _, condition := range node.Status.Conditions {
-		if condition.Type == corev1.NodeReady && condition.Status != corev1.ConditionTrue {
-			ready = false
-			break
+		if condition.Type == corev1.NodeReady {
+			return condition.Status == corev1.ConditionTrue
 		}
 	}
 	return ready
```
[the remark this answers is not in the document]
Comment (r0hansaxena) [reply pkg/utils/kubeclient/node.go]: I've switched to explicit bool return with explicit false default.

**Comment by r0hansaxena** [comment]
@cheyang could you please have a look again? thank you

**Comment by cheyang** [comment]
Important behavior fix: `IsReady()` previously returned `true` for nodes without a `NodeReady` condition, which is incorrect — a node with no readiness signal should be treated as not-ready. The refactored logic is cleaner (find-and-return vs flag-and-break) and the new tests cover all edge cases: empty conditions, missing NodeReady, ConditionUnknown, and mixed condition lists.

Verdict: **ready-to-merge**
